integration_test/test_videoroom/test/integration/simulcast_test.exs

```diff
@@ -0,0 +1,372 @@
+        @simulcast_inbound_stats,
+        @simulcast_inbound_stats,
+        @simulcast_inbound_stats,
+        @simulcast_inbound_stats
+      ]
+      |> Enum.zip(timeouts)
+
+    stages = [
+      :disable_medium,
+      :disable_high,
+      :disable_low,
+      :enable_low,
+      :enable_high,
+      :enable_medium
+    ]
+
+    buttons_with_id =
+      [buttons1, buttons2]
+      |> Enum.map(&Enum.zip(&1, stages))
+      |> Enum.with_index()
+      |> Map.new(fn {button, browser_id} -> {browser_id, button} end)
+
+    for {browser_id, buttons} <- buttons_with_id, into: [] do
+      start_buttons = if browser_id == 1, do: @start_buttons, else: []
+
+      specific_mustang = %{
+        mustang_options
+        | id: browser_id,
+          buttons: buttons,
+          start_buttons: start_buttons
+      }
+
+      Task.async(fn ->
+        Stampede.start({SimulcastMustang, specific_mustang}, @browser_options)
+      end)
+    end
+    |> Task.await_many(:infinity)
+
+    receive do
+      acc ->
+        for stage <- stages,
+            browsers = Map.get(acc, stage) do
+          encoding =
+            case stage do
+              :disable_medium -> "h"
+              :disable_high -> "l"
+              :disable_low -> nil
+              :enable_low -> "l"
+              :enable_high -> "h"
+              :enable_medium -> "m"
+            end
+
+          sender = browsers[0]
+          receiver = browsers[1]
+
+          stats = Enum.zip(sender, receiver)
+
+          case stage do
+            :disable_low ->
+              assert(
+                Enum.any?(receiver, &(&1["framesPerSecond"] == 0)),
+                "Failed on stage: #{stage} should be all encodings disabled,
+                receiver stats are #{inspect(receiver)}
+                sender stats are #{inspect(sender)}
+                "
+              )
+
+            _other ->
+              assert(
+                Enum.any?(stats, fn {sender, receiver} ->
+                  stats_for_encoding?(receiver, sender, encoding)
+                end),
+                "Failed on stage: #{stage} should be encoding: #{encoding},
+                receiver stats are: #{inspect(receiver)}
+                sender stats are: #{inspect(Enum.map(sender, & &1[encoding]))}
+                "
+              )
+          end
+        end
+    end
+  end
+
+  defp create_room(mustang_options) do
+    # Creating room earlier to avoid error :already_started
+    Task.async(fn ->
+      Stampede.start({RoomMustang, mustang_options}, @browser_options)
+    end)
+    |> Task.await(:infinity)
+
+    Process.sleep(2_000)
+  end
+
+  defp stats_for_encoding?(receiver, sender_stats, encoding) do
+    sender = sender_stats[encoding]
+
+    correct_dimensions? =
+      receiver["height"] == sender["height"] and receiver["width"] == sender["width"]
+
+    (correct_dimensions? or sender["qualityLimitationReason"] != "none") and
+      receiver["callbackEncoding"] == encoding
+  end
```

Comment:
I lost myself here. What is the difference between `sender_stats` and `sender`? Also, is `receiver` something analogous to `sender_stats`? If yes, why it is not called `receiver_stats`? 

I also feel like this function should be called in a different way. Something like `assert_stats_for_encoding` or `check_stats_for_encoding`. Having said that I would move checking stats for receiver to a separate function.

Reply:
As these are simulcast stats, so the sender has separate stats for each encoding. That's why the sender variable is `sender_stats[encoding`. 

The sender and receiver have some common keys but also have some different keys specific to each of them. 

We are checking stats for receiver and sender in the same function to be sure that the media received by the receiver has got the same dimension as the media sent by the sender for chosen encoding.

I think about changing names like this:
stats_for_encoding? -> check_stats_for_encoding?
sender_stats -> all_encoding_sender_stats
receiver -> receiver_stats
sender -> sender_stats

This way `receiver_stats` and `sender_stats` will be analogous because they will contain stats from one transceiver.

Reply:
I see. Current version is much better. However, I think that this function shouldn't get `encoding` as an argument. Instead, it should take `receiver_stats`, `sender_stats`  and check whether they are the same. We could then name this function as `assert_stats_equal`